parsing/__init__.py

```diff
@@ -1080,13 +713,16 @@ def __repr__(self):
         ret = "\n".join(lines)
         return ret
 
-    def _prepare(self, modules, pickleFile, pickleMode, logFile, graphFile):
+    def _prepare(self, adapter, pickleFile, pickleMode, logFile, graphFile):
         """
 Compile the specification into data structures that can be used by
 the Parser class for parsing.
 """
         # Get the grammar specification.
-        self._introspect(modules)
+        if isinstance(adapter, types.ModuleType) or (
```

Comment:
Let's fail with a `ValueError` here if `adapter` is neither a module, nor a list of modules, nor an object implementing `get_{precedences|tokens|nonterminals}`.